kork/kork-artifacts/src/test/java/com/netflix/spinnaker/kork/artifacts/artifactstore/s3/S3ArtifactStoreStorerTest.java

```diff
@@ -28,47 +28,42 @@
 import com.netflix.spinnaker.kork.artifacts.artifactstore.ArtifactStoreURISHA256Builder;
 import com.netflix.spinnaker.kork.artifacts.model.Artifact;
 import com.netflix.spinnaker.kork.common.Header;
-import com.netflix.spinnaker.kork.exceptions.SpinnakerException;
 import com.netflix.spinnaker.security.AuthenticatedRequest;
 import java.util.List;
 import java.util.stream.Stream;
-import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.core.sync.RequestBody;
 import software.amazon.awssdk.services.s3.S3Client;
-import software.amazon.awssdk.services.s3.model.HeadObjectRequest;
-import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.services.s3.model.*;
 
 public class S3ArtifactStoreStorerTest {
   @Test
-  public void testExceptionPathOfObjectExists() {
+  public void testObjectExistsPath() {
     S3Client client = mock(S3Client.class);
-    when(client.headObject(any(HeadObjectRequest.class)))
-        .thenThrow(S3Exception.builder().statusCode(400).build());
+    when(client.listObjectsV2(any(ListObjectsV2Request.class)))
+        .thenReturn(ListObjectsV2Response.builder().build());
+
+    verify(client, times(0)).putObject(any(PutObjectRequest.class), any(RequestBody.class));
     AuthenticatedRequest.set(Header.APPLICATION, "my-application");
     S3ArtifactStoreStorer artifactStoreStorer =
         new S3ArtifactStoreStorer(client, "my-bucket", new ArtifactStoreURISHA256Builder(), null);
-    String expectedExceptionMessage = "Failed to query artifact due to invalid request";
```

Comment:
I'm not seeing any failure tests right now on failure handling.  It's relatively minor, but we've lost a few of the exception handling test cases where or whatever reason, s3 blows up.

Reply:
Ah I see now for some reason the change in the s3storer didn't show up tracing through this.  Ignore this as you've refactored the code on this.